packages/nextjs/utils/scaffold-eth/fetchPriceFromUniswap.ts

```diff
@@ -1,15 +1,26 @@
-import { Fetcher, Route, Token, WETH } from "@uniswap/sdk";
+import { Fetcher, Route, Token } from "@uniswap/sdk";
 import { Provider } from "@wagmi/core";
+import { getTargetNetwork } from "~~/utils/scaffold-eth";
 
 export const fetchPriceFromUniswap = async (provider: Provider): Promise<number> => {
-  try {
-    const DAI = new Token(1, "0x6B175474E89094C44Da98b954EedeAC495271d0F", 18);
-    const pair = await Fetcher.fetchPairData(DAI, WETH[DAI.chainId], provider);
-    const route = new Route([pair], WETH[DAI.chainId]);
-    const price = parseFloat(route.midPrice.toSignificant(6));
-    return price;
-  } catch (error) {
-    console.error("useEthPrice - Error fetching ETH price from Uniswap: ", error);
+  const configuredNetwork = getTargetNetwork();
+  if (configuredNetwork.nativeCurrency.symbol == "ETH" || configuredNetwork.tokenAddress) {
+    try {
+      const DAI = new Token(1, "0x6B175474E89094C44Da98b954EedeAC495271d0F", 18);
+      const TOKEN = await Fetcher.fetchTokenData(
+        1,
+        configuredNetwork.tokenAddress ? configuredNetwork.tokenAddress : "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
```

Comment:
`configuredNetwork.tokenAddress || "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"` is a more concise/clean way to write it.

Reply:
Oooo nice something new to learn. Thanks